config/initializers/omniauth.rb

```diff
@@ -18,8 +18,25 @@
 
 Rails.application.config.middleware.use OmniAuth::Builder do
   issuer = ENV.fetch('OPENID_CONNECT_ISSUER', '')
+  lb = ENV.fetch('LOADBALANCER_ENDPOINT', '')
 
-  if issuer.present?
+  if lb.present?
+    provider :openid_connect, setup: lambda { |env|
+      request = Rack::Request.new(env)
+      current_provider = request.params['current_provider']
```

Comment:
Why not infer the current_provider from the request object?
This may duplicate the `parse_user_domain` logic (which we can DRY if that's the intent) but makes the request the only source of truth for such value and It's somehow more secure (this doesn't mean that the proposed solution is insecure).
We may remove the ENV API in the future,

Reply:
I see what you mean,

However what captured my intention is the fact that `request.params['current_provider']` equals the value of the hidden input field.
It's true that it's supposed to match the ENV API result when the page loads which equals to the `current_provider` from the API perspective.

It's true that a request headers can be easily manipulated but I thought that the request body is a bit easier to manipulate plus the flow will involve more steps and therefore more room for issues (to get the provider value instead of parsing the host header the value must propagated in a ping pong style with no verification). 

It's just a note not a recommendation,
I'm okay with both scenarios,